src/ironoak/person/persondetector.py

```diff
@@ -0,0 +1,113 @@
+
+from pathlib import Path
+import cv2
+import depthai as dai
+import numpy as np
+import time
+import argparse
+import threading
+
+class persondetector:
+    def __init__(self):
+        self.isdetected = False
+
+        self.labelMap = ["background", "aeroplane", "bicycle", "bird", "boat", "bottle", "bus", "car", "cat", "chair", "cow",
+            "diningtable", "dog", "horse", "motorbike", "person", "pottedplant", "sheep", "sofa", "train", "tvmonitor"]
+
+        nnPathDefault = str((Path(__file__).parent / Path('../models/mobilenet-ssd_openvino_2021.4_6shave.blob')).resolve().absolute())
+        parser = argparse.ArgumentParser()
+        parser.add_argument('nnPath', nargs='?', help="Path to mobilenet detection network blob", default=nnPathDefault)
+        parser.add_argument('-ff', '--full_frame', action="store_true", help="Perform tracking on full RGB frame", default=False)
```

Comment:
ArgumentParser() is for scripts that are run from the terminal. This is not required for a class that will called from elsewhere.
an example invocation:
pd_object = PersonDetector()
But, this does not trigger the ArgumentParser(). So only defaults are chosen.